components/constellation/constellation.rs

```diff
@@ -800,6 +800,12 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
+    fn add_pending_change(&mut self, change: SessionHistoryChange)
+    {
+        self.handle_load_start_msg(change.new_pipeline_id);
+        self.pending_changes.push(change);
```

Comment:
I wonder if we should accept all the properties of `SessionHistoryChange` in the method and create the instance inside this function.

`fn add_pending_change(&mut self, top_level_browsing_context_id: TopLevelBrowsingContextId, browsing_context_id: BrowsingContextId, pipeline_id: PipelineId, load_data: LoadData, replace: Option<Instant>`

I'm not sure if its much of an improvement though, the important part is that we call `self.handle_load_start_msg`